gossip/src/duplicate_shred_handler.rs

```diff
@@ -21,10 +25,8 @@ const MAX_NUM_CHUNKS: u8 = 3;
 // is only 1 person sending out duplicate proofs, 1 person is leader for 4 slots,
 // so we allow 5 here to limit the chunk map size.
 const ALLOWED_SLOTS_PER_PUBKEY: usize = 5;
-// To prevent an attacker inflating this map, we discard any proof which is too
-// far away in the future compared to root.
-const MAX_SLOT_DISTANCE_TO_ROOT: Slot = 100;
-// We limit the pubkey for each slot to be 100 for now.
+// We limit the pubkey for each slot to be 100 for now, when this limit is reached,
+// we drop 50% of pubkeys with lowest stakes.
 const MAX_PUBKEY_PER_SLOT: usize = 100;
```

Comment:
We probably want number of pubkeys _after_ pruning 50% does at least have 2/3 of stake.
Base on mainnet stake distribution right now, seems like this number should be more like ~300.

Reply:
done